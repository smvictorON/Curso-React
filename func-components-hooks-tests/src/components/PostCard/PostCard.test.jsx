import { render, screen } from '@testing-library/react';
import { PostCard } from '.';

const mock = {
  title: 'Titulo',
  body: 'Corpo',
  id: 1,
  cover: 'img/img.png',
};

describe('<PostCard/>', () => {
  it('should render poscard correctly', () => {
    render(<PostCard {...mock} />);
    // usado para mostrar o component no console
    // const { debug } = render(<PostCard {...mock} />);
    // debug();

    expect(screen.getByRole('img', { name: 'Titulo' })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: 'Titulo' })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: 'Titulo' })).toBeInTheDocument();
    expect(screen.getByText('Corpo')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...mock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
