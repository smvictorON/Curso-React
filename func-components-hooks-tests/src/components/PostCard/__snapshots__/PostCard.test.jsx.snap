// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Titulo"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Titulo
    </h2>
    <p>
      Corpo
    </p>
  </div>
</div>
`;
